SingularityUI/app/components/taskSearch/TaskSearch.cjsx

```diff
@@ -0,0 +1,198 @@
+Utils = require '../../utils'
+Enums = require './Enums'
+TaskSearchForm = require './TaskSearchForm'
+DisplayResults = require './DisplayResults'
+
+TaskSearch = React.createClass
+
+    headerText: 'Task Search'
+
+    countChoices: [5, 10, 25, 50]
+
+    countDefault: 10
+
+    getInitialState: ->
+        return {
+            requestId: @props.initialRequestId
+            deployId: @props.initialDeployId
+            host: @props.initialHost
+            lastTaskStatus: @props.initialTaskStatus
+            startedBefore: @props.initialStartedBefore
+            startedAfter: @props.initialStartedAfter
+            sortDirection: @props.initialSortDirection or 'ASC'
+            pageNumber: 1
+            count: @props.initialCount or @countDefault
+            showForm: true
+        }
+
+    handleSubmit: (event) ->
+        event.preventDefault()
+        @setState({
```

Comment:
more of a space-saving/readability thing we can do with coffeescript
```
@setState showForm: false
```
is the same as
```
 @setState({
      showForm: false
})
```